common/src/service/HttpClientHelper.cs

```diff
@@ -71,18 +75,19 @@ internal sealed class HttpClientHelper : IHttpClientHelper
                 _httpClientHandler.Proxy = customHttpProxy;
             }
 
+            _connectionLeaseTimeoutHandler = new ConnectionLeaseTimeoutHandler(connectionLeaseTimeout, _httpClientHandler);
```

Comment:
The essence of this change is to insert a small delegating handler who periodically inserts the "connectionClose" header on a request to force the http client to close the TCP connection after the request. The subsequent request will open a new TCP connection, but only after doing a DNS lookup first.

Reply:
This is sort of a breaking change, as users will now see a new TCP connection spawned occasionally, but it is infrequent enough that I don't expect any complaints.

Reply:
As long as we spell out what the impact of the change is in release notes, we should be fine.

Reply:
However, it looks like you took out the ability for a client to override our value. Seems like we should make it configurable. Perhaps they want faster than 5 minute detection.

Reply:
I went ahead and added the ability to configure these timeout again, but with the default still changing to 5 minutes rather than the infinite timeout we had before